ksql-common/src/main/java/io/confluent/ksql/util/KsqlConfig.java

```diff
@@ -189,115 +240,208 @@
           LogMetricAndContinueExceptionHandler.class
       );
     }
-
-    applyStreamsConfig(originals(), streamsConfigOverlay);
-    this.ksqlStreamConfigProps = ImmutableMap.copyOf(streamsConfigOverlay);
+    final Map<String, ResolvedConfig> streamConfigProps = new HashMap<>();
+    applyStreamsConfig(streamsConfigDefaults, streamConfigProps);
+    applyStreamsConfig(originals(), streamConfigProps);
+    this.ksqlStreamConfigProps = ImmutableMap.copyOf(streamConfigProps);
   }
 
-  private KsqlConfig(final Map<String, ?> values,
-                     final ImmutableMap<String, Object> ksqlStreamConfigProps) {
-    super(CONFIG_DEF, values);
+  private KsqlConfig(final boolean current,
+                     final Map<String, ?> values,
+                     final ImmutableMap<String, ResolvedConfig> ksqlStreamConfigProps) {
+    super(configDef(current), values);
     this.ksqlStreamConfigProps = ksqlStreamConfigProps;
   }
 
-  public ImmutableMap<String, Object> getKsqlStreamConfigProps() {
-    return ksqlStreamConfigProps;
+  public Map<String, Object> getKsqlStreamConfigProps() {
+    final Map<String, Object> props = new HashMap<>();
+    for (ResolvedConfig config : ksqlStreamConfigProps.values()) {
+      props.put(config.key, config.value);
+    }
+    return Collections.unmodifiableMap(props);
   }
 
-  public ImmutableMap<String, Object> getKsqlAdminClientConfigProps() {
-    return ksqlStreamConfigProps.entrySet().stream()
-        .map(Map.Entry::getKey)
-        .filter(AdminClientConfig.configNames()::contains)
-        .collect(
-            ImmutableMap.toImmutableMap(name -> name, ksqlStreamConfigProps::get)
-        );
+  public Map<String, Object> getKsqlAdminClientConfigProps() {
+    final Map<String, Object> props = new HashMap<>();
+    ksqlStreamConfigProps.values().stream()
+        .filter(resolvedConfig -> AdminClientConfig.configNames().contains(resolvedConfig.key))
+        .forEach(
+            resolvedConfig -> props.put(resolvedConfig.key, resolvedConfig.value));
+    return Collections.unmodifiableMap(props);
   }
 
-  public ImmutableMap<String, Object> getAllProps() {
-    final ImmutableMap.Builder<String, Object> builder = ImmutableMap.builder();
-    builder.putAll(values().entrySet().stream()
-        .filter(e -> e.getValue() != null)
-        .collect(
-            Collectors.toMap(
-                Map.Entry::getKey,
-                Map.Entry::getValue))
+  public Map<String, String> getKsqlConfigPropsCleaned() {
+    final Map<String, String> props = new HashMap<>();
+    configDef(true).names().stream().forEach(
+        key -> props.put(key, ConfigDef.convertToString(values().get(key), typeOf(key)))
     );
-    builder.putAll(ksqlStreamConfigProps.entrySet().stream()
-        .filter(e -> e.getValue() != null)
-        .collect(
+    return Collections.unmodifiableMap(props);
+  }
+
+  public Map<String, String> getKsqlStreamConfigPropsCleaned() {
```

Comment:
A comment explaining what cleaning is actually done would help. It seems like it's just doing a string conversion.

Reply:
Right, would be good to leave that as a code comment is my point.

Reply:
or perhaps name the method so that it is obvious what it is doing